ui/src/App.tsx

```diff
@@ -67,64 +73,51 @@ const App = () => {
     `;
 
   const DEFAULT_DATE = formatDate(new Date());
-
-  const [startDate, setStartDate] = useState<string>(DEFAULT_DATE);
-  const [endDate, setEndDate] = useState<string>(DEFAULT_DATE);
-  const [startTime, setStartTime] = useState<string>(DEFAULT_START_TIME);
-  const [endTime, setEndTime] = useState<string>(DEFAULT_END_TIME);
-  const [title, setTitle] = useState<string>("");
-  const [error, setError] = useState<string | null>(null);
   const [events, setEvents] = useState<EventSourceInput>([]);
   const [displayedEventData, setDisplayedEventData] = useState(
     {} as DisplayedEventData
   );
   const [showOverlay, setShowOverlay] = useState<boolean>(false);
+  const [inEditMode, setInEditMode] = useState<boolean>(false);
 
   useEffect(() => {
     getEntries().then((entries) => {
       setEvents(entries);
     });
   }, []);
 
-  const handleCreateEntry = async () => {
+  const handleCreateEntry = async ({
+    title,
+    startDate,
+    endDate,
+    startTime,
+    endTime,
+  }: formEntryProps) => {
     const startTimeUtc = new Date(`${startDate}T${startTime}`);
     const endTimeUtc = new Date(`${endDate}T${endTime}`);
-    createEntry({
+    await createEntry({
       title,
       startTimeUtc,
       endTimeUtc,
     });
-  };
-
-  const handleSubmit = (e: MouseEvent<HTMLButtonElement>) => {
-    e.preventDefault();
-    const startDateAndTime: string = `${startDate}T${startTime}`;
-    const endDateAndTime: string = `${endDate}T${endTime}`;
-    if (startDateAndTime > endDateAndTime) {
-      setError("Error: end cannot be before start.");
-      return;
-    }
-    handleCreateEntry();
     getEntries().then((entries) => {
       setEvents(entries);
     });
-
-    setTitle("");
-    setError(null);
-    setStartDate(DEFAULT_DATE);
-    setEndDate(DEFAULT_DATE);
-    setStartTime(DEFAULT_START_TIME);
-    setEndTime(DEFAULT_END_TIME);
   };
 
-  const showEventOverlay = (arg: EventClickArg) => {
-    const entryId = arg.event._def.extendedProps._id;
+  const getEntryDetails = (entryId: string) => {
     getEntry(entryId).then((data) => {
       setDisplayedEventData(data);
       setShowOverlay(true);
+      setInEditMode(false);
     });
   };
 
+  const showEventOverlay = (arg: EventClickArg) => {
```

Comment:
As discussed, let's change this to `openModal` or `openEventModal` so it's more clear what this function is doing in comparison to `setShowOverlay`